Hybrid Image, new flow factor schedules, optical flow guidance

reallybigname opened the pull request:
new option to use init image as hybrid video
- you can use all normal compositing stuff with that mode.
- you can't use optical flow, because it's not a video - but - if there are already video frames in the directory, well - you should be able to still use flow and ransac modes, although I haven't tested that

changed Optical flow redo to a selection of optical flow type with None as default
- added Redo flow factor schedule
- made optical flow redo look nicer by warping the same image that it is using to compare to prev_img, then it goes in for final generation changed Optical flow cadence to a selection of optical flow type with None as default
- added cadence flow factor schedule

Added flow guidance from last flow
- no UI change - it just works by default with hybrid motion optical flow types. It just uses the last flow as guidance, so it's a minor difference, but a little better temporal coherence on flow

changed hybrid composite to a selection with None as default
- now can set hybrid compositing to Normal (After Motion), Before Motion, or After Generation!

reallybigname committed:
Hybrid Image, new flow factor schedules, optical flow guidance

new option to use init image as hybrid video
- you can use all normal compositing stuff with that mode.
- you can't use optical flow, because it's not a video - but - if there are already video frames in the directory, well - you should be able to still use flow and ransac modes, although I haven't tested that

changed Optical flow redo to a selection of optical flow type with None as default
- added Redo flow factor schedule
- made optical flow redo look nicer by warping the same image that it is using to compare to prev_img, then it goes in for final generation
changed Optical flow cadence to a selection of optical flow type with None as default
- added cadence flow factor schedule

Added flow guidance from last flow
- no UI change - it just works by default with hybrid motion optical flow types. It just uses the last flow as guidance, so it's a minor difference, but a little better temporal coherence on flow

changed hybrid composite to a selection with None as default
- now can set hybrid compositing to Normal (After Motion), Before Motion, or After Generation!

reallybigname committed:
Fixed accidental newb error is not 'None'

I was using: is not 'None'
instead of: != 'None'

Ugh...

hithereai committed:
add deprecation updates

hithereai committed:
ui updates

hithereai committed:
more ui updates

hithereai committed:
remove old commented line

hithereai committed:
Merge remote-tracking branch 'upstream/automatic1111-webui' into pr/529

hithereai committed:
Merge branch 'fix_interp_ffmpeg' into pr/529

hithereai committed:
deduplicate gradio code

hithereai committed:
clean

hithereai committed:
decapitalize a word in gr.info

hithereai commented:
![image](https://user-images.githubusercontent.com/121192995/230341573-98ca2bef-b3ca-4898-b159-738b57963183.png)

hithereai committed:
print to printf fix

hithereai committed:
add comments

reallybigname committed:
Made hybrid_use_init_image option override the args.use_init_image in args,py

This change prevents the args.init_image from being changes to None in args.py is hybrid_use_init_image is on.

reallybigname committed:
Merge branch 'automatic1111-webui' of https://github.com/reallybigname/deforum-for-automatic1111-webui into automatic1111-webui

reallybigname committed:
Changed hybrid_use_init_image feature to work with online images

reallybigname commented:
Fixed to use online images properly with hybrid_use_init_image option. That should wrap it up as far as I know.

reallybigname commented:
It has come to my attention that the new hybrid use init image as video option is confusing as far as what you should expect when loading defaults. So, I've ran videos for each scenario. Personally, I think it's rather obvious when you just test the features, but I do use Deforum a lot, and have an intimate understand of it's quirks related to the first frame vs rest of the frames.

First, we have the situation of just loading defaults, then:

- Clicking Normal hybrid generate, and clicking Use init image as video
Because I never engaged the use_init_image option on the Init tab, the init image is not used for the first frame. But, you can see that it gets used for the rest of the frames:
https://user-images.githubusercontent.com/6529731/230472931-7edbf6db-ca34-4abe-8430-cc71a70aeafd.mp4

- Same settings, but enabling Use init image in Init tab
Now, you see the init image on the first frame
https://user-images.githubusercontent.com/6529731/230473483-08beb224-ec60-4707-95e7-90638552186b.mp4

- Clicking Normal hybrid generate, and clicking Use init image as video, and also clicking Generate Inputframes
Now, since the option to Use the first frame as init image is on by default, and inputframes are generated, it uses the first frame from the video as the init image and as the image for all frames
https://user-images.githubusercontent.com/6529731/230473854-4915f6ab-5ad0-4d16-8483-4e0b7be66557.mp4

- And finally, the same but with the new hybrid Use init image as video option disabled, and I turned off generate inputframes
You get a normal hybrid video, because inputframes are already generated.
https://user-images.githubusercontent.com/6529731/230474205-302fbbdb-c229-4721-ba66-e7dfb5472a61.mp4

It should be noted that once inputframes are generated, you can still use them for motion options, even when using the new Use init image as video option.

Don't hesitate to contact me if there are any other issues or clarifications needed. Happy to help - **AS ALWAYS**.

-Forest

hithereai committed:
Merge remote-tracking branch 'upstream/automatic1111-webui' into pr/529

hithereai committed:
Merge remote-tracking branch 'upstream/automatic1111-webui' into pr/529

hithereai committed:
auto-hide Cadence flow factor schedule

hithereai committed:
add ui info for optical_flow_cadence

hithereai committed:
better ui text

hithereai committed:
better ui text
